modules/authentication/proc-enabling-user-authentication-with-microsoft-azure.adoc

```diff
@@ -0,0 +1,491 @@
+[id="enabling-user-authentication-with-microsoft-azure"]
+= Enabling user authentication with {azure-brand-name}
+
+To authenticate users with {azure-brand-name}, configure the {azure-short} authentication provider in {product} and provision the users and groups from {azure-short} to the {product-short} software catalog.
+
+.Prerequisites
+* You have the permission to register an application in {azure-short}.
+Alternatively, you can ask your {azure-short} administrator to prepare the required {azure-short} application.
+
+* You link:{configuring-book-url}[added a custom {product-short} application configuration], and have sufficient permissions to modify it.
+
+* Your {product-short} backend can access the following hosts:
+
+`login.microsoftonline.com`::
+This is the {azure-brand-name} authorization server, which enables the authentication flow.
+
+`graph.microsoft.com`::
+For retrieving organization data, including user and group data, to be ingested into the {product-short} catalog.
+
+.Procedure
+:my-product-app-name-in-azure: <Authenticating with {product-short}>
+. To allow {product-short} to authenticate with {azure-short}, link:https://learn.microsoft.com/en-us/entra/identity-platform/scenario-web-app-sign-user-app-registration?tabs=aspnetcore#register-an-app-by-using-the-azure-portal[Register an app by using the {azure-short} portal].
+
+.. Sign in to the link:https://entra.microsoft.com/[Microsoft Entra admin center].
+
+.. Optional: If you have access to multiple tenants, use the *Settings* icon in the top menu to switch to the tenant in which you want to register the application from the *Directories + subscriptions* menu.
+
+.. Browse to *Applications > App registrations*, and create a **New registration** with the configuration:
+
+Name::
+Enter a name to identify your application in {azure-short}, such as __{my-product-app-name-in-azure}__.
+
+Supported account types::
+Select *Accounts in this organizational directory only*.
+
+Redirect URI::
+
+Select a platform:::
+Select *Web*.
+
+URL:::
+Enter the backend authentication URI set in {product-short}: `pass:c,a,q[{my-product-url}/api/auth/microsoft/handler/frame]`
+
+.. On the *Applications > App registrations > __{my-product-app-name-in-azure}__ > Manage > API permissions* page, *Add a Permission*, *Microsoft Graph*, select the following permissions:
+
+Application Permissions::
+`GroupMember.Read.All`:::
```

Comment:
Reading https://github.com/backstage/backstage/blob/201add3d17ef6aecc9f769d27939f233ced2edb3/plugins/catalog-backend-module-msgraph/README.md we might need this:

```suggestion
`GroupMember.Read.All`:::
`User.Read.All`:::
```

Can you confirm @JessicaJHee @kim-tsao that we have to add the `User.Read.All` permission twice, and that the delegated permission would not be enough?

Reply:
Yes - the Delegated permission (for the user signing in) should have `User.Read`.

The application permission (for the backend catalog plugin) needs `User.Read.All`.